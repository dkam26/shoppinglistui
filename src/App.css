.App {
 
  margin-left: 500px; 
  margin-right: 400px;
  margin-bottom: 40px;  
  border-style: solid;
  border-width: 5px;
  border-color: pink;
  width:400px;
  height: 600px;
  background-size: 400px 600px;
  background-image: url("static/Login.jpg");
  padding: 100px;
  background-repeat:no-repeat;
}
.Appregister {
  
   margin-left: 500px; 
   margin-right: 400px;
   margin-bottom: 40px;  
   border-style: solid;
   border-width: 5px;
   border-color: pink;
   width:400px;
   height: 600px;
   background-size: 400px 600px;
   background-image: url("static/Register.jpg");
   padding: 118px;
   background-repeat:no-repeat;
 }
.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}
.shoppinglists{
  
}
.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}
#xx{
  text-align: center;
  width:200px;
  margin-top: 4px;
  margin-left: -20px;
}
#x{
  text-align: center;
  width:200px;
  margin-top: 70px;
  margin-left: 72px;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
