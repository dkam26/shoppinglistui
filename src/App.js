import React, { Component } from 'react';
import './App.css';
import {Switch, Route} from 'react-router-dom';
import Login from "./components/home.component";
import Register from "./components/register.component";
import Shoppinglists from "./components/shoppinglists.component";
import Editshoppinglist from "./components/edit.shoppinglist.component";
import Addshoppinglist from "./components/add.shoppinglist.component";
import Items from "./components/items.component";
import EditItem from "./components/edit.item.component";
import AddItem from "./components/add.item.component";
class App extends Component {
  render() {
    return (
        <Switch id='switch' >
          <Route exact path="/" render={props =><Login {...props}/>}/>
          <Route path="/register"  render={props =><Register {...props}/>}/>
          <Route path="/shoppinglists"  render={props =><Shoppinglists {...props}/>}/>
          <Route  path="/editshoppinglist/:name"  render={props =><Editshoppinglist {...props}/>}/>
          <Route path="/addshoppinglist/"  render={props =><Addshoppinglist {...props}/>}/>
          <Route path="/items/:name"  render={props =><Items {...props}/>}/>
          <Route path="/editItem/:itemshoppinglist/:product/:amount/:quantity"  render={props =><EditItem  {...props}/>}/>
          <Route path="/addItem/:name"  render={props =><AddItem {...props}/>}/>
        </Switch>
    );
  }
}

export default App;
