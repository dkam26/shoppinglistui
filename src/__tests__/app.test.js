import React from 'react';
import { shallow, mount } from 'enzyme';
import {configure} from 'enzyme';
import App  from '../App';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import  toJson,{ shallowToJson } from 'enzyme-to-json';
import LocalStorageMock from '../setupTests';
import ReactDOM from 'react-dom';
configure({ adapter: new Adapter() });
describe('app component', () => {
    const props={
    
        render:()=>{}
     }
    const wrapper =  shallow(<App  />);
    it('should run', () => {
        expect(wrapper.exists(<div></div>)).toBe(true);
    });
    it('should ve the following method',()=>{
        wrapper.instance().render();
       
       
       });
    it('should render correctly', () => {  
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        });
})