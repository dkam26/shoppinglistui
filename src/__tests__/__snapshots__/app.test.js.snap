// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component should render correctly 1`] = `
<Switch
  id="switch"
>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    path="/register"
    render={[Function]}
  />
  <Route
    path="/shoppinglists"
    render={[Function]}
  />
  <Route
    path="/editshoppinglist/:name"
    render={[Function]}
  />
  <Route
    path="/addshoppinglist/"
    render={[Function]}
  />
  <Route
    path="/items/:name"
    render={[Function]}
  />
  <Route
    path="/editItem/:itemshoppinglist/:product/:amount/:quantity"
    render={[Function]}
  />
  <Route
    path="/addItem/:name"
    render={[Function]}
  />
</Switch>
`;

exports[`app component should ve the following method 1`] = `
<Switch
  id="switch"
>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    path="/register"
    render={[Function]}
  />
  <Route
    path="/shoppinglists"
    render={[Function]}
  />
  <Route
    path="/editshoppinglist/:name"
    render={[Function]}
  />
  <Route
    path="/addshoppinglist/"
    render={[Function]}
  />
  <Route
    path="/items/:name"
    render={[Function]}
  />
  <Route
    path="/editItem/:itemshoppinglist/:product/:amount/:quantity"
    render={[Function]}
  />
  <Route
    path="/addItem/:name"
    render={[Function]}
  />
</Switch>
`;
